src/collections/maple_tree/mod.rs

```diff
@@ -370,159 +505,185 @@ where
+        if distance == 0 {
             return true;
         }
-        let freePivots = R_CAPACITY-self.len();
+        let freePivots = R_CAPACITY - self.len();
 
         // make sure there is enough room to move the pivots into
-        if freePivots < places{
+        if freePivots < distance {
             return false;
         }
-        // make sure the pivot is greater than 0 and less than the index of the 
+        // make sure the pivot is greater than 0 and less than the index of the
         // last pivot as moving that would push it out of the array
-        else if pivotIndex < 0 || pivotIndex >= R_CAPACITY-1{
+        else if pivotIndex < 0 || pivotIndex >= R_CAPACITY - 1 {
             return false;
-            //panic!("PivotIndex isn't within the correct bounds"); 
-        }
-        else if pivotIndex+places > R_CAPACITY{
+        //panic!("PivotIndex isn't within the correct bounds");
+        } else if pivotIndex + distance > R_CAPACITY {
             return false;
-            //panic!("the number of places to move the pivots was to large, must be bellow {}", R_CAPACITY); 
+            //panic!("the number of distance to move the pivots was to large, must be bellow {}", R_CAPACITY);
         }
 
         // must use a variable to hold the pivots and values between swaps as we can't have two
         // mutable borrows of the same array at the same time :(
         let mut pivotHolder: M<K> = M::None;
         let mut valueHolder: M<V> = M::None;
-        let lastIndex = (R_CAPACITY)-freePivots;
-        
-        // starts at lastPivotIndex and goes until reaching the pivot at pivotIndex 
-        // so moving pivots and values from righ to left so each pivot and value is moved 
+        let lastIndex = (R_CAPACITY) - freePivots;
+
+        // starts at lastPivotIndex and goes until reaching the pivot at pivotIndex
+        // so moving pivots and values from righ to left so each pivot and value is moved
         // into a free place in the arrays
-        for i in 0..lastIndex+1{
-            if lastIndex-i >= pivotIndex {
-                match &self.pivot[lastIndex-i]{
+        for i in 0..lastIndex + 1 {
+            if lastIndex - i >= pivotIndex {
+                match &self.pivot[lastIndex - i] {
                     M::None => {
-                        continue; 
-                    },
-                    M::Some(_) => {  
-                            mem::swap(&mut pivotHolder, &mut self.pivot[lastIndex-i]);
-                            mem::swap(&mut pivotHolder, &mut self.pivot[lastIndex-i+places]);
-
-                            mem::swap(&mut valueHolder, &mut self.value[lastIndex-i]);
-                            mem::swap(&mut valueHolder, &mut self.value[lastIndex-i+places]);
+                        continue;
+                    }
+                    M::Some(_) => {
+                        mem::swap(&mut pivotHolder, &mut self.pivot[lastIndex - i]);
```

Comment:
Apparently there is a &mut slice method for swap:

https://doc.rust-lang.org/std/primitive.slice.html#method.swap

So you can do it iplace rather than memswapping in and out.

Reply:
awesome, I will use this then!